### PR TITLE
Add time selector for hearings 

### LINKED ISSUE #8653: Add time selector for hearings
## User Stories: 
* As a hearing coordinator, I will sometimes need the ability to select a time that is different than or more specific than 8:30/12:30 for video hearings or 9:00/1:00 for CO hearings. 
* As a hearing coordinator, if I will need to select a specific hearing time in half hour increment for Veterans for future Board telehearings.
* As a hearing coordinator, I may need to select a specific hearing time to accommodate the needs of alternate hearing locations. 

## Mocks: 
* Daily Docket - https://va-gov.invisionapp.com/share/H3PXTJOA4DX 
screen shot 2019-01-14 at 1 39 14 pm
screen shot 2019-01-14 at 1 39 27 pm

* Schedule Veteran modal - https://va-gov.invisionapp.com/share/K6PYN2NBTQF
screen shot 2019-01-14 at 1 39 55 pm
screen shot 2019-01-14 at 1 40 15 pm

* Reschedule Veteran modal  - https://va-gov.invisionapp.com/share/XUPVE1WEGTR#/339762030_Daily_Docket_Schedule_Now (page 2 for module, behavior should mimic the Schedule Veteran module)
screen shot 2019-01-14 at 1 39 41 pm

## Acceptance Criteria 
- [ ] Verify that a user can select "Other" radio button from time choices
- [ ] Verify that when "Other" is selected, user is presented with a drop down time selector that gives scheduling in 30 minute increments from 8:30 a.m. - 4:30 p.m.

## Related Ticket
See #8645 for the larger Daily Docket design updates.

### PR DESCRIPTION
**Pull request by Sjones352**
Resolves #8653

### Description
Pr adds time selectors to the modal in Case details and Daily Dockets

### Acceptance Criteria
- [ ] Verify that a user can select "Other" radio button from time choices
- [ ] Verify that when "Other" is selected, user is presented with a drop down time selector that gives scheduling in 30 minute increments from 8:30 a.m. - 4:30 p.m.

### Testing Plan
1. Go to Daily Docket

screen shot 2019-01-30 at 2 53 02 pm

Schedule Modal
screen shot 2019-01-30 at 3 37 58 pm

### CODE REVIEW

File: client/app/hearingSchedule/components/DailyDocket.jsx [hunk 1]
```diff
@@ -238,12 +250,22 @@ export default class DailyDocket extends React.Component {
...
-      hideLabel /></div>;
+    <div {...radioButtonStyling}>
+      <RadioField
+        name= "time"
+        label= "Time"
+        options={this.getHearingTimeOptions(hearing, readOnly)}
+        value={hearing.editedTime ? hearing.editedTime : getTimeWithoutTimeZone(hearing.scheduledFor, timezone)}
+        onChange={this.onHearingTimeUpdate(hearing.id)}
+        strongLabel />
+      <SearchableDropdown
```
Comment (mdbenjam) [inline client/app/hearingSchedule/components/DailyDocket.jsx]: Are we supposed to hide this unless they choose other?

File: client/app/hearings/constants/constants.js [hunk 2]
```diff
@@ -68,3 +68,34 @@ export const SERVER_ERROR_CODES = {
...
+  label: '11:00' },
+{ value: '11:30',
+  label: '11:30' },
+{ value: '12:00',
+  label: '12:00' },
+{ value: '12:30',
+  label: '12:30' },
+{ value: '1:00',
+  label: '1:00' },
+{ value: '2:00',
```
Comment (mdbenjam) [inline client/app/hearings/constants/constants.js]: Is there intentionally not a 1:30 hearing?

File: client/app/hearingSchedule/containers/DailyDocketContainer.jsx [hunk 3]
```diff
@@ -140,6 +140,28 @@ export class DailyDocketContainer extends React.Component {
       });
   };
 
+  getOptionalTime = (hearing) => {
```
Comment (mdbenjam) [inline client/app/hearingSchedule/containers/DailyDocketContainer.jsx]: Where is this method being called from?

File: client/app/hearings/constants/constants.js (hunk 2 above)
Comment (mdbenjam) [reply client/app/hearings/constants/constants.js]: If you'd like we can generate this list programatically.

File: client/app/queue/components/AssignHearingModal.jsx [hunk 4]
```diff
@@ -1,4 +1,7 @@
 // @flow
+/* eslint no-mixed-spaces-and-tabs: 0*/
```
Comment (mdbenjam) [inline client/app/queue/components/AssignHearingModal.jsx]: I don't think we should be disabling these. Can we make sure there are no tabs in this file? You can search for tabs by doing a search for `\t`

File: client/app/queue/components/AssignHearingModal.jsx [hunk 5]
```diff
@@ -254,7 +273,7 @@ class AssignHearingModal extends React.PureComponent<Props, LocalState> {
   };
 
   formatHearingDate = () => {
-    const { selectedHearingDay, selectedHearingTime } = this.props;
+    const { selectedHearingDay, selectedHearingTime, selectedOptionalTime } = this.props;
 
     if (selectedHearingDay && !selectedHearingTime) {
```
[the remark this answers is not in the document]
Comment (mdbenjam) [reply client/app/queue/components/AssignHearingModal.jsx]: Could you rephrase this? Not sure I understand what you mean.

File: client/app/components/common/actions.js [hunk 6]
```diff
@@ -49,3 +49,10 @@ export const onReceiveDropdownData = (dropdownName, data) => ({
     data
   }
 });
+
+export const onHearingOptionalTime = (optionalTime) => ({
```
Comment (mdbenjam) [inline client/app/components/common/actions.js]: This is fine, you could also try to use the same action, `onHearingTimeChange` for both changes. Maybe by passing in objects
```
{
  radioButton: 'Other',
  dropDown: '10:30'
}
```

File: client/app/queue/components/AssignHearingModal.jsx (hunk 4 above)
Comment (mdbenjam) [reply client/app/queue/components/AssignHearingModal.jsx]: Were you able to find any tabs?

File: client/app/hearingSchedule/containers/DailyDocketContainer.jsx [hunk 7]
```diff
@@ -87,7 +88,7 @@ export class DailyDocketContainer extends React.Component {
   };
 
   getTime = (hearing) => {
-    if (hearing.editedTime) {
+    if (hearing.editedTime || hearing.selectedOptionalTime) {
```
Comment (mdbenjam) [inline client/app/hearingSchedule/containers/DailyDocketContainer.jsx]: Is this right? Seems like if `selectedOptionalTime` is set we want to get the hour and minute from `selectedOptionalTime` instead of `editedTime`?

File: client/app/queue/components/AssignHearingModal.jsx [hunk 8]
```diff
@@ -254,29 +277,26 @@ class AssignHearingModal extends React.PureComponent<Props, LocalState> {
   };
 
   formatHearingDate = () => {
-    const { selectedHearingDay, selectedHearingTime } = this.props;
+    const { selectedHearingDay, selectedHearingTime, selectedOptionalTime } = this.props;
 
-    if (selectedHearingDay && !selectedHearingTime) {
+    if (!selectedHearingTime && !selectedOptionalTime) {
```
Comment (mdbenjam) [inline client/app/queue/components/AssignHearingModal.jsx]: Is there a chance that `selectedHearingTime` is null? The old code checked for this. Want to make sure dropping that check was intentional.

File: client/app/queue/components/AssignHearingModal.jsx [hunk 9]
```diff
@@ -286,12 +306,26 @@ class AssignHearingModal extends React.PureComponent<Props, LocalState> {
       hearingDate: hearingDay.hearingDate,
       regionalOffice: this.getRO()
     };
+  };
+
+  setSelectedTime = (val) => {
+  	const selectedTime = _.isString(val) ? val : val.value;
```
Comment (mdbenjam) [inline client/app/queue/components/AssignHearingModal.jsx]: I think these are the tabs. The spacing is off.

File: client/app/hearingSchedule/components/AssignHearings.jsx [hunk 10]
```diff
@@ -63,14 +63,15 @@ export default class AssignHearings extends React.Component {
...
+
     const { selectedRegionalOffice, selectedHearingDay } = this.props;
 
     // St. Petersburg, FL or Winston-Salem, NC
     if (selectedRegionalOffice === 'RO17' || selectedRegionalOffice === 'RO18') {
       return selectedHearingDay.room;
     }
 
-    return '';
+    return ' ';
```
Comment (mdbenjam) [inline client/app/hearingSchedule/components/AssignHearings.jsx]: Is this change intentional?

Comment (oscarAtNava) [reply client/app/hearingSchedule/components/AssignHearings.jsx]: No, I did not do this. May have come in when merging? @Sjones352 what is the correct line?

Comment (Sjones352) [reply client/app/hearingSchedule/components/AssignHearings.jsx]: @oscarAtNava Yes but fixed it now.
